salt/utils/event.py

```diff
@@ -200,7 +200,7 @@ def tagify(suffix='', prefix='', base=SALT):
         parts.extend(suffix)
     else:  # string so append
         parts.append(suffix)
-    return TAGPARTER.join([part for part in parts if part])
+    return TAGPARTER.join([str(part) for part in parts if part])
```

Comment:
This is going to break if any of the `parts` contains non ascii characters in a `unicode` type. This should be fixed before we invoke the join, not in a list comprehension. Something like this:

```python
for index, _ in enumerate(parts):
    try:
        parts[index] = salt.utils.stringutils.to_str(parts[index])
    except TypeError:
        continue
return TAGPARTER.join(parts)
```

Reply:
Since `salt.utils.stringutils.to_str(parts[index])` will raise TypeError for integer types, how about something like this to handle integers?
```python
    for index, _ in enumerate(parts):
        if isinstance(parts[index], six.integer_types):
            parts[index] = str(parts[index])
        else:
            try:
                parts[index] = salt.utils.stringutils.to_str(parts[index])
            except TypeError:
                pass
    return TAGPARTER.join(parts)
```